src/RequestModelFactory.php

```diff
@@ -27,17 +30,19 @@ public function __construct(ValidatorInterface $validator, Injector $injector, C
 
     /**
      * @param ServerRequestInterface $request
-     * @param array|ReflectionParameter[] $handlerParams
+     * @param ReflectionParameter[] $handlerParameters
      *
      * @throws ReflectionException
      *
      * @return array
      */
-    public function createInstances(ServerRequestInterface $request, array $handlerParams): array
+    public function createInstances(ServerRequestInterface $request, array $handlerParameters): array
     {
         $requestModelInstances = [];
-        foreach ($this->getModelRequestClasses($handlerParams) as $modelClass) {
-            $requestModelInstances[] = $this->processModel($request, $this->injector->make($modelClass));
+        foreach ($this->getModelRequestClasses($handlerParameters) as $modelClass) {
+            /** @var RequestModelInterface $modelInstance */
```

Comment:
Try in `getModelRequestClasses()` returns `class-string<RequestModelInterface>[]`. Maybe this phpdoc not need.

Reply:
This phpdoc for `$modelInstance` not `$modelClass`

Reply:
Yes, but injector support typed return for psalm: https://github.com/yiisoft/injector/blob/d936d15e8a0c70d3486832ad75edf73688d4c309/src/Injector.php#L107